app/models/api.rb

```diff
@@ -0,0 +1,11 @@
+class Api
+  def self.request(system, endpoint, args = nil)
+    "Api::#{Api.env}#{system.capitalize}::#{endpoint.capitalize}".constantize.new(*args).request
+  end
+
+  def self.env
+    if Rails.env == "development" || Rails.env == "test"
+      "FakeData::"
```

Comment:
This is a clever way of doing this, although I don't love a giant string + `constantize`.  Unfortunately, I can't think of a better way of doing it beyond like...matching the passed in strings against a hash with values that are the classes or something.  And I don't love that idea, either!

Reply:
i think because there are so many paths (fakedata vs no, system, endpoint) the hash would not be amazing, but i am happy to spec it out if it feels more readable! (or an alternative)

Reply:
does this feel any better?

```
class Api
  def self.request(system, endpoint, args = nil)
    Api
      .namespace
      .const_get(system.capitalize)
      .const_get(endpoint.capitalize)
	  .new(*args)
	  .request
  end

  def self.namespace
    if Rails.env == "development" || Rails.env == "test"
      Api::FakeData
    else
      Api
    end
  end
end
```

Reply:
I think I like this version better. It'd likely also make Brakeman happier.

Reply:
I will add a comment! what do you two think about having an endpoint validity check as well? would that be good for security? or is it Too Much(tm)

```
SYSTEMS = ["hses"]
ENDPOINTS = ["issue", "issues"]

def self.valid_request?(system, endpoint)
  SYSTEMS.include?(system) && ENDPOINTS.include?(endpoint)
end
```

Reply:
also this is a lot of methods now, some which take arguments. should i instantiate an instance?

Reply:
I think that would be a very good thing to do if we were allowing users to specify the system or endpoint. Since they're hard-coded in the controllers we should get failures of `const_get` when running unit tests if bad values are passed so I don't think it adds much.

Reply:
> also this is a lot of methods now, some which take arguments. should i instantiate an instance?

I don't think so, especially as long as the arguments stay on `self.request`.

Oh, also - we could avoid the `options` issue if we move that from the initializer to the request method.